Wikipedia/UI-V5/WMFSaveableTitleCollectionViewCell.m

```diff
@@ -0,0 +1,115 @@
+#import "WMFSaveableTitleCollectionViewCell+Subclass.h"
+
+#import "Wikipedia-Swift.h"
+#import "PromiseKit.h"
+#import "WMFSaveButtonController.h"
+#import "MWKTitle.h"
+
+#import "UIView+WMFShadow.h"
+#import "UIImageView+MWKImage.h"
+#import "UIButton+WMFButton.h"
+
+@interface WMFSaveableTitleCollectionViewCell ()
+
+@property (strong, nonatomic) WMFSaveButtonController* saveButtonController;
+
+@end
+
+@implementation WMFSaveableTitleCollectionViewCell
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super init];
+    if (self) {
+        [self commonInit];
+    }
+    return self;
+}
+
+- (void)awakeFromNib {
+    [super awakeFromNib];
+    [self commonInit];
+}
+
+- (void)commonInit {
+    [self configureContentView];
+    self.imageView.image = [UIImage imageNamed:[[self class] defaultImageName]];
+    [self.saveButton wmf_setButtonType:WMFButtonTypeBookmark];
+}
+
+- (void)configureContentView {
+    self.clipsToBounds               = NO;
+    self.contentView.backgroundColor = [UIColor whiteColor];
+    [self.contentView wmf_setupShadow];
+}
+
+- (void)prepareForReuse {
+    [super prepareForReuse];
+    self.title           = nil;
+    self.imageURL        = nil;
+    self.imageView.image = [UIImage imageNamed:[[self class] defaultImageName]];
+}
+
+#pragma mark - Accessors
+
++ (NSString*)defaultImageName {
+    return @"lead-default";
+}
+
+- (void)setSaveButton:(UIButton*)saveButton {
+    self.saveButtonController.button = saveButton;
+}
+
+- (UIButton*)saveButton {
+    return self.saveButtonController.button;
+}
+
+- (void)setSavedPageList:(MWKSavedPageList*)savedPageList {
+    self.saveButtonController.savedPageList = savedPageList;
```

Comment:
If this was wired at the view controller level, then the cell wouldn't need to know about the saved page list. 

Not saying the cell shouldn't/couldn't retain the reference to the savedButtonController for ease of use - just that cell needs to know about the saved page list only because it needs to set the property on the button controller.